plugins/scaffolder-backend/src/scaffolder/tasks/TaskWorker.test.ts

```diff
@@ -219,21 +303,38 @@ describe('TaskWorker internals', () => {
       },
     };
 
+    const subscribers = new Set<
+      ZenObservable.SubscriptionObserver<{ events: SerializedTaskEvent[] }>
+    >();
+
     let claimedTaskCount = 0;
     const taskWorker = new TaskWorkerConstructor({
       runners: { workflowRunner },
       taskBroker: {
+        event$() {
+          return new ObservableImpl<{ events: SerializedTaskEvent[] }>(
+            subscriber => {
+              subscribers.add(subscriber);
+              return () => {
+                subscribers.delete(subscriber);
+              };
+            },
+          );
+        },
         async claim() {
           claimedTaskCount++;
           return {
+            async getWorkspaceName() {
```

Comment:
Another side note here I think, is that getWorkspaceName _happens_ to return the task ID. It's not documented or settled behavior. Using it to deduce the task ID is risky and subject to future breakage.